tradeexecutor/analysis/trade_analyser.py

```diff
@@ -564,50 +564,53 @@ def add_prop(value, key: str, formatter: Callable):
         return df
 
     def show(self):
+        """Render a summary table in IPython notebook."""
+        self.show_custom(self.to_dataframe())
+
+    @staticmethod
+    def show_custom(df: pd.DataFrame):
         """Render a summary table in IPython notebook."""
         with pd.option_context("display.max_row", None):
-            df = self.to_dataframe()
             display(df.style.set_table_styles([{'selector': 'thead', 'props': [('display', 'none')]}]))
     
-    def get_full_report(self):
+    def check_quantstats(self):
+        """Check that requirements are met for using quantstats library. Used as decorator for provided helper methods."""
+        if not HAS_QUANTSTATS:
+            raise RuntimeError("Quantstats library not installed")
+
+        if not hasattr(self, "daily_returns"):
+            raise RuntimeError("Daily returns have not been calculated. Remember to provided state \
+                argument. E.g. summary = analysis.calculate_summary_statistics(state=state)")
+
+    def get_full_report(self) -> None:
         """Show basic and advanced stats and plots"""
-        return (
-            qs.reports.full(self.daily_returns) 
-            if HAS_QUANTSTATS and self.daily_returns is not None
-            else None
-        )
+        self.check_quantstats()
+        return qs.reports.full(self.daily_returns) 
     
-    def get_basic_stats(self):
-        """Show basic stats only"""
-        return (
-            qs.reports.metrics(self.daily_returns)
-            if HAS_QUANTSTATS and self.daily_returns is not None
-            else None
-        )
+    def get_basic_stats(self) -> pd.DataFrame:
+        """Gets basic stats only.
+        
+        returns: Pandas DataFrame object 
+        """
+        self.check_quantstats()
+        return qs.reports.metrics(self.daily_returns, display=False)
 
-    def get_full_stats(self):
-        """Show basic and advanced stats"""
-        return (
-            qs.reports.metrics(self.daily_returns, mode='full')
-            if HAS_QUANTSTATS and self.daily_returns is not None
-            else None
-        )
+    def get_full_stats(self) -> pd.DataFrame:
+        """Gets basic and advanced stats.
+        
+        returns: Pandas DataFrame object"""
+        self.check_quantstats()
+        return qs.reports.metrics(self.daily_returns, mode='full', display=False)
 
-    def get_basic_plots(self):
+    def get_basic_plots(self) -> None:
```

Comment:
Function cannot be called "get" and returning `None` so something is still amiss here

Reply:
have changed `get` to `show` for methods returning `None`